venues/ICLR.cc/2018/Workshop/python/tpms-dump.py

```diff
@@ -0,0 +1,136 @@
+import argparse
+import openreview
+import csv
+import config
+
+## Argument handling
+parser = argparse.ArgumentParser()
+parser.add_argument('csv', help="input file with emails, first_name, last_name")
+parser.add_argument('datestring', help="append to output file name")
+parser.add_argument('--baseurl', help="base url")
+parser.add_argument('--username')
+parser.add_argument('--password')
+args = parser.parse_args()
+
+client = openreview.Client(baseurl=args.baseurl, username=args.username, password=args.password)
+print "connecting to ", client.baseurl
+datestring = args.datestring
+
+# create mapping from email --> first,last
+names_by_email = {}
+
+with open(args.csv) as f:
+    for email, first, last in csv.reader(f):
+        names_by_email[email] = (first, last)
+
+print "retrieving profiles where available ..."
+email_by_signature = {}
+profile_by_email = {}
+for email, names in names_by_email.iteritems():
+    try:
+        # TODO - why does profile differ from client.get_note()
+        profile = client.get_profile(email)
```

Comment:
when you do "get profile", it calls the profiles API, which puts a mask on email addresses and other private information. The note gives unmasked information.

e.g.
profile: emails=["p****r@cs.umass.edu"]
note: emails=["pmandler@cs.umass.edu"]